### PR TITLE
Feature/interpolation

### PR DESCRIPTION
**Pull request by fccoelho**
Marcelo,

I created this PR to prepare your code for merging and facilitate code review

### CODE REVIEW

**Comment by fccoelho** [comment]
@marfcg  I think the first thing to be changed should be the directory name, instead of 'funs_marfcg' it should be something like "intepolation"

File: GeraAlerta/fun/funs_marfcg/missingdata.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python
```
Comment (fccoelho) [inline GeraAlerta/fun/funs_marfcg/missingdata.py]: Perhaps we should stick  to python3?

**Comment by fccoelho** [comment]
I think that the sample data could be moved into a subdir, and the actual unit tests could be written to use them and check for correct behavior. You can find simple examples of how to write tests here: https://docs.python.org/3/library/unittest.html#module-unittest

**Comment by marfcg** [comment]
Perfect. I'll implement the changes ASAP.
Also, as a second step, I'll work on the migration to pandas for ease of
code.

Thanks for the feedback!

---
*Sent from mobile device. My apologies for eventual typos.
Marcelo Ferreira da Costa Gomes
Programa de Computação Científica
Fundação Oswaldo Cruz
PROCC/FIOCRUZ

I think that the sample data could be moved into a subdir, and the actual
unit tests could be written to use them and check for correct behavior. You
can find simple examples of how to write tests here:
https://docs.python.org/3/library/unittest.html#module-unittest

—
Reply to this email directly or view it on GitHub
<https://github.com/fccoelho/AlertaDengueAnalise/pull/13#issuecomment-114061077>
.

**Comment by marfcg** [comment]
- Updates in commit 38557b34a3:
  - Folder renamed to 'interpolate'.
  - All python codes inside 'interpolate' now default to env python3. (PS: Should we make this change global? Must double check codes for compatibility, though. 'print', for instance, is a function in python3, not a statement as in python2.x). Tested OK on local machine with python3 + dependencies installed.
  - All sample files moved to sampledata dubdir.
This commit will be frozen (hotfixes aside) until the other two improvements are done.

- Work in progress (implemented as separate branches):
  - Unit test implementation (https://github.com/marfcg/AlertaDengueAnalise/tree/feature/unittest/interpolation)
  - Migration to Pandas (https://github.com/marfcg/AlertaDengueAnalise/tree/feature/migration2pandas/interpolation)

Unfortunately, I didn't start the interpolation branch as feature/interpolation/main
Using that convention, those two new branches could have been named feature/interpolation/migration2pandas and feature/interpolation/unittest for instance.

**Comment by marfcg** [comment]
Migration to pandas complete.
Main function for missing data estimation:
```python
def fill_missingdata(df, win):
    '''
    Fill missing data in df whenever df[df.index[1]].isnull() == True
    Uses smoothdata function

    Input:
    :df: pandas DataFrame with 2 columns. Assumes first column is the independent variable.
    :win: window used for data smoothing. 
          Check smoothdata function for documentation.

    Output:
    :df_filled: DataFrame with originally empty values substituted
                by estimated values.
    :df_smoothed: DataFrame used for (extra)interpolation
```

**Pull request closed without merging**